
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>CDS_Search &#8212; AdaptSearch 2 documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ConcatPhyl" href="ConcatPhyl.html" />
    <link rel="prev" title="Orthogroups_Tool" href="Orthogroups_Tool.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="ConcatPhyl.html" title="ConcatPhyl"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="Orthogroups_Tool.html" title="Orthogroups_Tool"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="cds-search">
<h1>CDS_Search<a class="headerlink" href="#cds-search" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>This tool takes files containing nucleic aligned sequences and search the ORF and the CDS.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By default, AdaptSearch will run with nucleic format, but you can choose to use proteic format from this point to run ConcatPhyl and CodeML.</p>
</div>
</div>
<div class="section" id="user-parameters">
<h2>User parameters<a class="headerlink" href="#user-parameters" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt>For computing :</dt>
<dd><ol class="first last arabic simple">
<li>(yes/no) Methionine : if ‘yes’ is checked (default), the Methionine will be considered in the search of CDS</li>
<li>(integer) Minimal number of species in each locus. It is recommended to set this number to the total number of studied species, to avoid missing data afterwards</li>
<li>(integer) Minimal length of the CDS (in amino-acids)</li>
<li>(integer) Minimal length of the subsequence (in amino-acids) between two series of indels</li>
<li>(integer) Minimal length of the CDS, (in nucleotides) without indels</li>
</ol>
</dd>
<dt>For outputs selection - All available in both nucleic and proteic format</dt>
<dd><ol class="first last arabic simple">
<li>Activate outputs for files with the best ORF</li>
<li>Activate outputs for files with CDS</li>
<li>Activate outputs for files with CDS without indels (most important)</li>
</ol>
</dd>
</dl>
</div>
<div class="section" id="code-documentation">
<h2>Code documentation<a class="headerlink" href="#code-documentation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="algorithm">
<h3>Algorithm<a class="headerlink" href="#algorithm" title="Permalink to this headline">¶</a></h3>
<div class="section" id="part-1">
<h4>Part. 1:<a class="headerlink" href="#part-1" title="Permalink to this headline">¶</a></h4>
<p>Predict potential ORF on the basis of 2 criteria + 1 optional criteria :</p>
<blockquote>
<div><ol class="arabic simple">
<li>Longest part of the alignment of sequence without codon stop “*”, tested in the 3 potential ORF</li>
<li>This longest part should be &gt; 150nc or 50aa</li>
<li>[OPTIONNAL] A codon start “M” should be present in this longuest part, before the last 50 aa<ol class="arabic">
<li>The output directory “05_CDS_aa” &amp; “05_CDS_nuc” does not include this criteria</li>
<li>The output directory “06_CDS_with_M_aa” &amp; “06_CDS_with_M_nuc” includes this criteria</li>
</ol>
</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="part-2">
<h4>Part. 2:<a class="headerlink" href="#part-2" title="Permalink to this headline">¶</a></h4>
<p>Find and removes too short regions or regions with too many indels, based on the user parameters ‘Minimal length of the CDS’, ‘Minimal length of the subsequence between two series of indels’ and ‘Minimal length of the CDS without indels’</p>
</div>
</div>
<div class="section" id="source-code">
<h3>Source code<a class="headerlink" href="#source-code" title="Permalink to this headline">¶</a></h3>
<div class="section" id="file-s01-find-orf-on-multiple-alignment-py">
<h4>File : S01_find_orf_on_multiple_alignment.py<a class="headerlink" href="#file-s01-find-orf-on-multiple-alignment-py" title="Permalink to this headline">¶</a></h4>
<dl class="function">
<dt id="code_universel">
<code class="descname">code_universel</code><span class="sig-paren">(</span><em>F1</em><span class="sig-paren">)</span><a class="headerlink" href="#code_universel" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>F1</strong> (<em>file</em>) – a txt file with the genetic code (each line : a codon and its amino-acid)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the genetic code (key : codon, value : amino-acid)</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Stop codons have the value ‘*’.</p>
<dl class="function">
<dt id="find_good_ORF_criteria_3">
<code class="descname">find_good_ORF_criteria_3</code><span class="sig-paren">(</span><em>bash_aligned_nc_seq</em>, <em>bash_codeUniversel</em><span class="sig-paren">)</span><a class="headerlink" href="#find_good_ORF_criteria_3" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>bash_aligned_nc_seq</strong> (<em>dict</em>) – A dictionary storing the sequences of the input fasta file (obtaiend by the function dico())</li>
<li><strong>bash_codeUniversel</strong> (<em>dict</em>) – Obtained with the function code_universel()</li>
<li><strong>minimal_cds_length</strong> (<em>int</em>) – the minimal length of a cds, in amino-acids</li>
<li><strong>min_spec</strong> (<em>int</em>) – the minimum number of species within the group after the cds search</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 dictionaries</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="docutils">
<dt>This is the most important fonction, which is run on each input file. The dictionaries return, in nucleic and proteic format :</dt>
<dd><ol class="first last arabic simple">
<li>All the sequences with their best ORF</li>
<li>All the sequences with their CDS without Methionine</li>
<li>All the sequences with their CDS with Methionine</li>
</ol>
</dd>
</dl>
<div class="section" id="find-good-orf-criteria-3-subroutines">
<h5>find_good_ORF_criteria_3() subroutines<a class="headerlink" href="#find-good-orf-criteria-3-subroutines" title="Permalink to this headline">¶</a></h5>
<dl class="function">
<dt id="multiple3">
<code class="descname">multiple3</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="headerlink" href="#multiple3" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> (<em>string</em>) – a rna/dna sequence</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">an updated sequence and the length(seq)%3 modulo</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">tuple</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Tests if a sequence is multiple of 3, which means only full codons. If not, the end of the sequence is cut by one or two bases.</p>
<dl class="function">
<dt id="simply_get_ORF">
<code class="descname">simply_get_ORF</code><span class="sig-paren">(</span><em>seq_dna</em>, <em>gen_code</em><span class="sig-paren">)</span><a class="headerlink" href="#simply_get_ORF" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq_dna</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>gen_code</strong> (<em>dict</em>) – Obtained with the function code_universel()</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">an amino-acid sequence translated from seq_dna (open-reading frame only)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">string</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="detect_Methionine">
<code class="descname">detect_Methionine</code><span class="sig-paren">(</span><em>seq_aa</em>, <em>Ortho</em>, <em>minimal_cds_length</em><span class="sig-paren">)</span><a class="headerlink" href="#detect_Methionine" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq_aa</strong> (<em>string</em>) – an amino-acid sequence</li>
<li><strong>Ortho</strong> (<em>int</em>) – value indicating wether the first methionine found is not in the last 50 amino-acids.</li>
<li><strong>minimal_cds_length</strong> (<em>int</em>) – the minimal length of a cds, in amino-acids</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">an updated value of <cite>Ortho</cite></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Detect if methionin in the aa sequence.</p>
<dl class="function">
<dt id="ReverseComplement2">
<code class="descname">ReverseComplement2</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="headerlink" href="#ReverseComplement2" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">The reverse complement of <cite>seq</cite></td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">string</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt>
<code class="descname">write_output_file(results_dict, name_elems, path_out):</code></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>results_dict</strong> (<em>dict</em>) – the results of the cds search on the input file</li>
<li><strong>name_elems</strong> (<em>list of Strings</em>) – a list containing all the words of the output file names. Edited with the orthogroup number and its number of species</li>
<li><strong>parth_out</strong> – the path to the output directory</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>
<div class="section" id="file-s02-remove-too-short-bit-or-whole-sequence-py">
<h4>File : S02_remove_too_short_bit_or_whole_sequence.py<a class="headerlink" href="#file-s02-remove-too-short-bit-or-whole-sequence-py" title="Permalink to this headline">¶</a></h4>
<dl class="function">
<dt id="detect_short_indel">
<code class="descname">detect_short_indel</code><span class="sig-paren">(</span><em>seq</em>, <em>MAX_LENGTH_SMALL_INDEL</em><span class="sig-paren">)</span><a class="headerlink" href="#detect_short_indel" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>MAX_LENGTH_SMALL_INDEL</strong> (<em>int</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">The lists of consecutive gap positions</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list of lists</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">detect_short_indel</span><span class="p">(</span><span class="s2">&quot;agcga-ag----agact-aca--ga-----aaatg-aca---aaaa&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="c1"># [[5], [17], [21, 22], [35]]</span>

<span class="n">detect_short_indel</span><span class="p">(</span><span class="s2">&quot;agcga-ag----agact-aca--ga-----aaatg-aca---aaaa&quot;</span><span class="p">,</span> <span class="mi">7</span><span class="p">)</span>
<span class="c1"># [[5], [8, 9, 10, 11], [17], [21, 22], [25, 26, 27, 28, 29], [35], [39, 40, 41]]</span>
</pre></div>
</div>
</div>
<div class="section" id="file-s03-remove-sties-with-not-enought-species-represented-py">
<h4>File : S03_remove_sties_with_not_enought_species_represented.py<a class="headerlink" href="#file-s03-remove-sties-with-not-enought-species-represented-py" title="Permalink to this headline">¶</a></h4>
<dl class="function">
<dt id="remove_position_with_too_much_missing_data">
<code class="descname">remove_position_with_too_much_missing_data</code><span class="sig-paren">(</span><em>bash_aa</em>, <em>bash_nuc</em>, <em>MIN_SPECIES_NB</em><span class="sig-paren">)</span><a class="headerlink" href="#remove_position_with_too_much_missing_data" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>bash_aa</strong> (<em>dict</em>) – a hashtable sequence_name - proteic sequence</li>
<li><strong>bash_nuc</strong> (<em>dict</em>) – a hashtable sequence_name - nucleic sequence</li>
<li><strong>MIN_SPECIES_NB</strong> (<em>int</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">two dictionaries without sequences containing indels in <cite>bash_aa</cite> and <cite>bash_nuc</cite></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p><cite>bash_aa</cite> and <cite>bash_nuc</cite> are obtained at each input file iteration with the function <cite>dico(fileIN)</cite> from <cite>dico.py</cite> file.</p>
<p>Scripts S02 and S03 are for Part 2 (indels removal).</p>
<p>Back to <a class="reference external" href="index.html">main page</a>.</p>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Filter_Assemblies.html">Filter_Assemblies</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pairwise.html">Pairwise</a></li>
<li class="toctree-l1"><a class="reference internal" href="POGs.html">POGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BlastAlign.html">BlastAlign</a></li>
<li class="toctree-l1"><a class="reference internal" href="TransDecoder.html">TransDecoder</a></li>
<li class="toctree-l1"><a class="reference internal" href="OrthoFinder_OnlyGroups.html">OrthoFinder_OnlyGroups</a></li>
<li class="toctree-l1"><a class="reference internal" href="Orthogroups_Tool.html">Orthogroups_Tool</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">CDS_Search</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#user-parameters">User parameters</a></li>
<li class="toctree-l2"><a class="reference internal" href="#code-documentation">Code documentation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#algorithm">Algorithm</a></li>
<li class="toctree-l3"><a class="reference internal" href="#source-code">Source code</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li class="toctree-l1"><a class="reference internal" href="MutCount.html">MutCount</a></li>
<li class="toctree-l1"><a class="reference internal" href="CodeML.html">codeML</a></li>
<li class="toctree-l1"><a class="reference internal" href="GeneSearch.html">GeneSearch</a></li>
<li class="toctree-l1"><a class="reference internal" href="Visualization_With_Python.html">Visualization with Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Shiny_Apps.html">Visualization with RShiny apps</a></li>
<li class="toctree-l1"><a class="reference internal" href="CommandLine_User_Guide.html">User Guide for the use of AdaptSearch in command line</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="Orthogroups_Tool.html"
                        title="previous chapter">Orthogroups_Tool</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="ConcatPhyl.html"
                        title="next chapter">ConcatPhyl</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/CDS_Search.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="ConcatPhyl.html" title="ConcatPhyl"
             >next</a> |</li>
        <li class="right" >
          <a href="Orthogroups_Tool.html" title="Orthogroups_Tool"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Alice Mataigne.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.1.
    </div>
  </body>
</html>
