
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>codeML &#8212; AdaptSearch 2 documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="GeneSearch" href="GeneSearch.html" />
    <link rel="prev" title="MutCount" href="MutCount.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="GeneSearch.html" title="GeneSearch"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="MutCount.html" title="MutCount"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="codeml">
<h1>codeML<a class="headerlink" href="#codeml" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>codeml is a part of the PAML package, which is a suite of programs for phylogenetic analyses of DNA or protein sequences using maximum likelihood (ML).</p>
<p>codeml carries out ML analysis of :</p>
<blockquote>
<div><ol class="arabic simple">
<li>protein-coding DNA sequences using codon substitution models (e.g., Goldman and Yang 1994)</li>
<li>amino acid sequences under a number of amino acid substitution models.</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="user-parameters">
<h2>User parameters<a class="headerlink" href="#user-parameters" title="Permalink to this headline">¶</a></h2>
<p>Full readme at <a class="reference external" href="http://abacus.gene.ucl.ac.uk/software/pamlDOC.pdf">http://abacus.gene.ucl.ac.uk/software/pamlDOC.pdf</a></p>
<p>The most important parameters are the model choices, ‘branch model’ and ‘Site model’. The two of them can be combined to run mixed models.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Incompatibilities between models are automatically excluded from the selection, but some can remain (we haven’t tested everything yet)!</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Don’t forget to set the ‘Sequences format in the fasta file’ advanced parameter according to the format of your alignement ! The AdaptSearch suite is configured to run with nucleic format by default, but it is still possible to perform analyses on proteic data (the tool CDS_Search can return proteic sequences, and ConcatPhyl can handle them)</p>
</div>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<p class="last">Some models require to edit the tree file (to set different ratios on branches) ; for now, there is no automatic way to do that : the user has to download the treeFile from Galaxy, edit it, and reload it in Galaxy. It could be interesting to dive more into the ete3 Python module, which implements a tree class, to see if there is an efficient way to edit trees.</p>
</div>
<div class="section" id="the-codeml-config-file">
<h3>The codeml config file<a class="headerlink" href="#the-codeml-config-file" title="Permalink to this headline">¶</a></h3>
<p>The program codeML reads all the model parameter from a pre-formatted configfile (‘codeml.ctl’), which is written within Galaxy, based on the parameters specified by the user. This config file looks like this :</p>
<p>The config file looks like this :</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>      seqfile = fastaAlignementFile * sequence data file name
      outfile = outputDirName * main result file name
     treefile = newickTreeFile * tree structure file name
        noisy = 9  * 0,1,2,3,9: how much rubbish on the screen
      verbose = 0  * 1: detailed output, 0: concise output
      runmode = 0  * 0: user tree;  1: semi-automatic;  2: automatic
                   * 3: StepwiseAddition; (4,5):PerturbationNNI; -2: pairwise
      seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs
    CodonFreq = 1  * 0:1/61 each, 1:F1X4, 2:F3X4, 3:codon table
*       ndata = 10
        clock = 0   * 0:no clock, 1:clock; 2:local clock; 3:TipDate
       aaDist = 0  * 0:equal, +:geometric; -:linear, 1-6:G1974,Miyata,c,p,v,a
                   * 7:AAClasses
   aaRatefile = wag.dat * only used for aa seqs with model=empirical(_F)
                  * dayhoff.dat, jones.dat, wag.dat, mtmam.dat, or your own
        model = 0
                   * models for codons:
                       * 0:one, 1:b, 2:2 or more dN/dS ratios for branches
                   * models for AAs or codon-translated AAs:
                       * 0:poisson, 1:proportional,2:Empirical,3:Empirical+F
                       * 6:FromCodon, 8:REVaa_0, 9:REVaa(nr=189)
      NSsites = 0  * 0:one w;1:neutral;2:selection; 3:discrete;4:freqs;
                   * 5:gamma;6:2gamma;7:beta;8:beta&amp;w;9:beta&amp;gamma;
                   * 10:beta&amp;gamma+1; 11:beta&amp;normal&gt;1; 12:0&amp;2normal&gt;1;
                   * 13:3normal&gt;0
        icode = 0  * 0:universal code; 1:mammalian mt; 2-11:see below
        Mgene = 0  * 0:rates, 1:separate;
    fix_kappa = 0  * 1: kappa fixed, 0: kappa to be estimated
        kappa = 2  * initial or fixed kappa
    fix_omega = 0  * 1: omega or omega_1 fixed, 0: estimate
        omega = 0.2 * initial or fixed omega, for codons or codon-based AAs
    fix_alpha = 1  * 0: estimate gamma shape parameter; 1: fix it at alpha
        alpha = 0.0 * initial or fixed alpha, 0:infinity (constant rate)
       Malpha = 0  * different alphas for genes
        ncatG = 3  * # of categories in dG of NSsites models
      fix_rho = 1  * 0: estimate rho; 1: fix it at rho
          rho = 0. * initial or fixed rho,   0:no correlation
        getSE = 0  * 0: don&#39;t want them, 1: want S.E.s of estimates
 RateAncestor = 0  * (0,1,2): rates (alpha&gt;0) or ancestral states (1 or 2)
   Small_Diff = 5e-07
    cleandata = 0  * remove sites with ambiguity data (1:yes, 0:no)?
  fix_blength = 0   * 0: ignore, -1: random, 1: initial, 2: fixed
       method = 0   * 0: simultaneous; 1: one branch at a time
</pre></div>
</div>
</div>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p>Publication : Yang, Z. (2007). PAML 4: Phylogenetic Analysis by Maximum Likelihood. In Molecular Biology and Evolution, 24 (8), pp. 1586–1591. [doi:10.1093/molbev/msm088]</p>
<p>Website and manual : <a class="reference external" href="http://abacus.gene.ucl.ac.uk/software/paml.html">http://abacus.gene.ucl.ac.uk/software/paml.html</a></p>
<p>Back to <a class="reference external" href="index.html">main page</a>.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Filter_Assemblies.html">Filter_Assemblies</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pairwise.html">Pairwise</a></li>
<li class="toctree-l1"><a class="reference internal" href="POGs.html">POGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BlastAlign.html">BlastAlign</a></li>
<li class="toctree-l1"><a class="reference internal" href="TransDecoder.html">TransDecoder</a></li>
<li class="toctree-l1"><a class="reference internal" href="OrthoFinder_OnlyGroups.html">OrthoFinder_OnlyGroups</a></li>
<li class="toctree-l1"><a class="reference internal" href="Orthogroups_Tool.html">Orthogroups_Tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="CDS_Search.html">CDS_Search</a></li>
<li class="toctree-l1"><a class="reference internal" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li class="toctree-l1"><a class="reference internal" href="MutCount.html">MutCount</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">codeML</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#user-parameters">User parameters</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#the-codeml-config-file">The codeml config file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="GeneSearch.html">GeneSearch</a></li>
<li class="toctree-l1"><a class="reference internal" href="Visualization_With_Python.html">Visualization with Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Shiny_Apps.html">Visualization with RShiny apps</a></li>
<li class="toctree-l1"><a class="reference internal" href="CommandLine_User_Guide.html">User Guide for the use of AdaptSearch in command line</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="MutCount.html"
                        title="previous chapter">MutCount</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="GeneSearch.html"
                        title="next chapter">GeneSearch</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/CodeML.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="GeneSearch.html" title="GeneSearch"
             >next</a> |</li>
        <li class="right" >
          <a href="MutCount.html" title="MutCount"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Alice Mataigne.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.1.
    </div>
  </body>
</html>
