
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>ConcatPhyl &#8212; AdaptSearch 2 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="MutCount" href="MutCount.html" />
    <link rel="prev" title="CDS_Search" href="CDS_Search.html" /> 
  </head>
  <body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="MutCount.html" title="MutCount"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="CDS_Search.html" title="CDS_Search"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="concatphyl">
<h1>ConcatPhyl<a class="headerlink" href="#concatphyl" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>This tool takes files containing fasta sequences (from CDS_Search outputs in the AdaptSearch suite, in nucleic or proteic format), concatenate all the sequences in a super-alignment, and run RAxML to build a phylogeny.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By default, AdaptSearch will run with nucleic format, but you can choose to use proteic format from CDS_Search to run ConcatPhyl and CodeML.</p>
</div>
</div>
<div class="section" id="user-parameters">
<h2>User parameters<a class="headerlink" href="#user-parameters" title="Permalink to this headline">¶</a></h2>
<div class="section" id="inputs">
<h3>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><ul class="simple">
<li>Files from Filter Assemblies : a set of fasta files (one file per species), e.g. the outputs of the first tool of the AdaptSearch suite. Used to retrieve all the species names.</li>
<li>Alignment files without indels : a set of fasta files with aligned sequences (with the same species than into the previous parameter), e.g the outputs of the CDS_Search tool of the AdaptSearch suite.</li>
<li>Format choice : nucleic or proteic (according to the CDS_Search output you choose).</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="raxml-parameters">
<h3>RAxML parameters<a class="headerlink" href="#raxml-parameters" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><ul class="simple">
<li>Substitution model (-m) : odel of Binary (Morphological), Nucleotide, Multi-state, or Amino-Acid substitution. Default : GTRGAMMA (nucleic), PROTCAT (proteic)</li>
<li>Matrix : AA substitution model (only when using proteic inputs). Default : DAYHOFF</li>
<li>random seed : Specifies a random number seed for the parsimony inferences. For all options/algorithms in RAxML that require some sort of randomization, this option must be specified. Make sure to pass different random number seeds to RAxML and not only 12345</li>
<li>Number of runs (-N) : Specifies the number of alternative runs. Default : 100</li>
<li>Use bootstopping criteria for number of runs : if selected, overxwrites the number of runs to use bootstopping criteria</li>
<li>Algorithm to execute (-f) : allows to choose what kind of algorithme RAxML shall execute. Default : Rapid bootsrap and best ML tree search (-f a)</li>
<li><dl class="first docutils">
<dt>Multiple model assignement to alignment partitions (-q) <span class="classifier-delimiter">:</span> <span class="classifier">an optional parameter. Permits to specify the file name which contains the assignment of models to alignment partitions for multiple models of substitution. For the syntax of this file please consult the manual.</span></dt>
<dd>This option allows you to specify the regions of your alignment for which an individual model of nucleotide substitution should be estimated. This will typically be useful to infer trees for long multi-gene alignments.</dd>
</dl>
</li>
<li>Rapid bootstrapping random seed (-x) : Specify an integer number (random seed) and turn on rapid bootstrapping. In addition to the best tree search. By default, this option is choosen.</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="outputs">
<h3>Outputs<a class="headerlink" href="#outputs" title="Permalink to this headline">¶</a></h3>
<p>The output files vary according to the</p>
<blockquote>
<div><ul class="simple">
<li>Phylogeny : the general output. It gives the information about the concatenation (statistics) and the RAxML run.</li>
</ul>
</div></blockquote>
<div class="section" id="concatenation-files">
<h4>Concatenation files<a class="headerlink" href="#concatenation-files" title="Permalink to this headline">¶</a></h4>
<p>The file name varies according to format parameters :</p>
<blockquote>
<div><ul class="simple">
<li>Phylogeny_concatenation_fasta_aa : sequences concatenated in fasta format when proteic format is chosen</li>
<li>Phylogeny_concatenation_phylip_aa : sequences concatenated in phylip format when proteic format is chosen</li>
<li>Phylogeny_concatenation_nexus_aa : sequences concatenated in nexus format when proteic format is chosen</li>
<li>Phylogeny_concatenation_fasta_nuc : sequences concatenated in fasta format when nucleic format is chosen</li>
<li>Phylogeny_concatenation_phylip_nuc : sequences concatenated in phylip format when nucleic format is chosen. <strong>This is the output used for the RAxML run</strong></li>
<li>Phylogeny_concatenation_nexus_nuc : sequences concatenated in nexus format when nucleic format is chosen</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="raxml-outputs">
<h4>RAxML outputs<a class="headerlink" href="#raxml-outputs" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><ul class="simple">
<li>Phylogeny_RAxML_BestTree** : the Best Tree found.</li>
<li>Phylogeny_RAxML_BiPartitionBranchLabel : the Best Tree found with supported values as branch labels.</li>
<li>Phylogeny_RAxML_BiPartition : the Best Tree found with supported values.</li>
<li>Phylogeny_RAxML_BootStrap : all the boostrapped trees. The number of boostraped trees depending of the option -N (number of run).</li>
</ul>
</div></blockquote>
</div>
</div>
</div>
<div class="section" id="code-documentation">
<h2>Code documentation<a class="headerlink" href="#code-documentation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="algorithm">
<h3>Algorithm<a class="headerlink" href="#algorithm" title="Permalink to this headline">¶</a></h3>
<p>See publication :</p>
<p>Stamatakis A. RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics. 2014;30(9):1312-1313. doi:10.1093/bioinformatics/btu033.</p>
</div>
<div class="section" id="source-code">
<h3>Source code<a class="headerlink" href="#source-code" title="Permalink to this headline">¶</a></h3>
<p>The python script, ‘S01_concatenate.py’ paste all the aligned orthogroups in a super-alignment, in fasta, phylip or nexus format</p>
<dl class="function">
<dt id="dico">
<code class="descname">dico</code><span class="sig-paren">(</span><em>F2</em><span class="sig-paren">)</span><a class="headerlink" href="#dico" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>F2</strong> (<em>String</em>) – A fasta file (aligned orthogroup) name</td>
</tr>
<tr class="field-even field"><th class="field-name">Return dicoco:</th><td class="field-body">a dictionary of sequences in the orthogroup, headers being the keys and sequences being the values</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>This function is called on each input orthogroup fasta file in the function concatenate()</p>
<dl class="function">
<dt id="concatenate">
<code class="descname">concatenate</code><span class="sig-paren">(</span><em>L_IN</em>, <em>SPECIES_ID_LIST</em><span class="sig-paren">)</span><a class="headerlink" href="#concatenate" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>L_IN</strong> (<em>list</em>) – A list of aligned orthogroups fasta files</li>
<li><strong>SPECIES_ID_LIST</strong> – The list of all studied species (abbreviated names)</li>
<li><strong>SPECIES_ID_LIST</strong> – list</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">bash_concat (dict) - the dictionary of the super-alignment</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">ln_concat (int) - the length of the super-alignment</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">nb_locus (int) - the number of concatenated orthogroups</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">list_genes_position (list of list) - the list of positions (with names of implied files) where each concatenantion is made</p>
</td>
</tr>
</tbody>
</table>
<p>This function builds a dictionary with all the species as key. The values are nucleic or proteic sequences, pasted alltogether along with iterations over the files in L_IN. ‘-‘ are inserted in a concatened sequence when a species is missing in a group.</p>
</dd></dl>

<dl class="function">
<dt id="get_codon_position">
<code class="descname">get_codon_position</code><span class="sig-paren">(</span><em>seq_inORF</em><span class="sig-paren">)</span><a class="headerlink" href="#get_codon_position" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq_inORF</strong> (<em>String</em>) – a nucleic sequence</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">4 Strings, storing the letter at a given position of each codon - the first one contains the first letter of each codons, the second contains the second letter of each codon, the third contains the first and second letters of each codons, the fourth contains the 3rd letter of each codon</td>
</tr>
</tbody>
</table>
</dd></dl>

<div class="highlight-python"><div class="highlight"><pre><span></span><span class="n">seq</span><span class="o">=</span><span class="s1">&#39;catgctagctagagaact&#39;</span>
<span class="n">a</span><span class="p">,</span><span class="n">b</span><span class="p">,</span><span class="n">c</span><span class="p">,</span><span class="n">d</span> <span class="o">=</span> <span class="n">get_codon_position</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>
<span class="k">print</span> <span class="n">a</span><span class="p">,</span><span class="n">b</span><span class="p">,</span><span class="n">c</span><span class="p">,</span><span class="n">d</span>

<span class="o">&gt;</span> <span class="n">cgataa</span> <span class="n">acgagc</span> <span class="n">cagcagtaagac</span> <span class="n">ttcgat</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p>RAxML publication : Stamatakis A. RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics. 2014;30(9):1312-1313. doi:10.1093/bioinformatics/btu033.</p>
<p>RAxML Manual : <a class="reference external" href="https://sco.h-its.org/exelixis/resource/download/NewManual.pdf">https://sco.h-its.org/exelixis/resource/download/NewManual.pdf</a></p>
<p>Back to <a class="reference external" href="index.html">main page</a>.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Filter_Assemblies.html">Filter_Assemblies</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pairwise.html">Pairwise</a></li>
<li class="toctree-l1"><a class="reference internal" href="POGs.html">POGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BlastAlign.html">BlastAlign</a></li>
<li class="toctree-l1"><a class="reference internal" href="TransDecoder.html">TransDecoder</a></li>
<li class="toctree-l1"><a class="reference internal" href="OrthoFinder_OnlyGroups.html">OrthoFinder_OnlyGroups</a></li>
<li class="toctree-l1"><a class="reference internal" href="Orthogroups_Tool.html">Orthogroups_Tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="CDS_Search.html">CDS_Search</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">ConcatPhyl</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#user-parameters">User parameters</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#inputs">Inputs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#raxml-parameters">RAxML parameters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#outputs">Outputs</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#code-documentation">Code documentation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#algorithm">Algorithm</a></li>
<li class="toctree-l3"><a class="reference internal" href="#source-code">Source code</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="MutCount.html">MutCount</a></li>
<li class="toctree-l1"><a class="reference internal" href="CodeML.html">codeML</a></li>
<li class="toctree-l1"><a class="reference internal" href="CodeML_Results_Analysis.html">Analysis of codeML results</a></li>
<li class="toctree-l1"><a class="reference internal" href="Visualization_With_Python.html">Visualization with Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Shiny_Apps.html">Visualization with RShiny apps</a></li>
<li class="toctree-l1"><a class="reference internal" href="CommandLine_User_Guide.html">User Guide for the use of AdaptSearch in command line</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="CDS_Search.html"
                        title="previous chapter">CDS_Search</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="MutCount.html"
                        title="next chapter">MutCount</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/ConcatPhyl.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="MutCount.html" title="MutCount"
             >next</a> |</li>
        <li class="right" >
          <a href="CDS_Search.html" title="CDS_Search"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Victor Mataigne.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.2.
    </div>
  </body>
</html>