
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>GeneSearch &#8212; AdaptSearch 2 documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Visualization with Python" href="Visualization_With_Python.html" />
    <link rel="prev" title="codeML" href="CodeML.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="Visualization_With_Python.html" title="Visualization with Python"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="CodeML.html" title="codeML"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="genesearch">
<h1>GeneSearch<a class="headerlink" href="#genesearch" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>These scripts call functions from the python module ETE Toolkit and have been developped to work in two ways :</p>
<div class="section" id="handling-of-codeml-runs-on-a-super-alignment-from-adaptsearch-s-galaxy-instance">
<h3>Handling of codeML runs on a super-alignment from AdaptSearch’s Galaxy instance<a class="headerlink" href="#handling-of-codeml-runs-on-a-super-alignment-from-adaptsearch-s-galaxy-instance" title="Permalink to this headline">¶</a></h3>
<p>This way of working is for analysis on the super-alignement obtained from the tool Concatphyl, on which the user can have ran several model of codeML. The python script receives all the results files and directory and proceeds to run model comparisons by Maximum Likelihood analysis. For now, all the results have to be downloaded from Galaxy, which can be really boring ; in a close future, we could add the script to the Galaxy suite</p>
</div>
<div class="section" id="running-an-evolutionary-model-on-many-differents-alignments-at-once">
<h3>Running an evolutionary model on many differents alignments at once :<a class="headerlink" href="#running-an-evolutionary-model-on-many-differents-alignments-at-once" title="Permalink to this headline">¶</a></h3>
<p>This way have been developped in the case the user wants to test evolutionary hypothesis on the aligned orthogroups before the concatenation into a super-alignement by the tool Concatphyl. The script receives all the alignement fasta files and run the same model on each of them. Results are stored into directories, one per input file. Ideally, it should be possible to run several models and to perform ML comparison on each model (per file)</p>
<ul class="simple">
<li>For now, The tree file is the tree computed by the super-alignement !</li>
<li><strong>The orthogroups must be complete (all studied species within) !</strong></li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This script can also work to run models on the super-alignment produced by Concatphyl. Maybe we should consider to implement it under Galaxy. Indeed, the ete3 Python module incorporates the whole PAML suite, with pre-built models, access to all parameters, and various explanations</p>
</div>
</div>
</div>
<div class="section" id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h2>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<p class="last">Some models require to edit the tree file (to set different ratios on branches) ; for now, there is no automatic way to do that : the user has to download the treeFile from Galaxy, edit it, and reload it in Galaxy. It could be interesting to dive more into the ete3 Python module, which implements a tree class, to see if there is an efficient way to edit trees.</p>
</div>
<div class="admonition-todo admonition" id="index-1">
<p class="first admonition-title">Todo</p>
<p class="last">There is currently a unfinished Galaxy’s wrapper.</p>
</div>
<div class="section" id="images-outputs-or-gui-s-visualisation">
<h3>Images outputs or GUI’s visualisation<a class="headerlink" href="#images-outputs-or-gui-s-visualisation" title="Permalink to this headline">¶</a></h3>
<p>ete3 allows the export of trees and alignments with branch and sites annotated with dn/ds values and selection detection.</p>
<div class="admonition-todo admonition" id="index-2">
<p class="first admonition-title">Todo</p>
<p class="last">This part is actually not implemented due to conflicts between Python versions, ete versions, and PyQt versions. The existing Python scripts have few commented commands unachieved.</p>
</div>
</div>
<div class="section" id="script-for-exploitation-of-codeml-results-load-precomputed-model-py">
<h3>Script for exploitation of codeML results : load_precomputed_model.py<a class="headerlink" href="#script-for-exploitation-of-codeml-results-load-precomputed-model-py" title="Permalink to this headline">¶</a></h3>
<p>The arguments are the RAxML best tree and the super-alignment computed by Concatphyl, and a text file with the list of models computed by codeML (either under Galaxy or without Galaxy).</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span># command line :
./load_precomputed_model.py /path/to/phylogenetic/tree/used/in/codeML /path/to/super/alignment/used/in/codeML /path/to/file/with/list/of/models
# Example with test-data files :
./scripts/load_precomputed_model.py test-data/concatenation_4_species.fasta test-data/tree_4_species.nwk test-data/pre_computed_models/models_list


# models list file should be like that (one model per line) :
M0 /path/to/M0/Model/out
M1 /path/to/M1/Model/out
M2 /path/to/M2/Model/out
fb /path/to/fb/Model/out

etc. The &#39;out&#39; file in the path is the main codeml output, which can have another name (unde Galaxy, it is named *run_codeml*)
</pre></div>
</div>
<p>All the models are charged in memory, and then all possible ML ratios tests are performed to determine the best model. The Summary table is printed on screen/log file and written into a csv file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">ML tests between two models are possible if :
#. The alternative model has more parameters than the null model
#. The difference log_likelihood(null_model) - log_likelihood(alternative model) must be under 0 (log_likelihoods are always negative numbers, so Log_likelihood(null_model) &lt; log_likelihood(alternative model))
#. For implementation purpose, <strong>the models names must be different</strong> : if you ran two models M0 with different parameters, name them something like ‘M0_a’ and ‘M0_b’. This is a point to improve in the future.</p>
</div>
</div>
<div class="section" id="script-for-running-codeml-models-on-one-or-many-alignments-load-precomputed-model-py">
<h3>Script for running codeML models on one or many alignments : load_precomputed_model.py<a class="headerlink" href="#script-for-running-codeml-models-on-one-or-many-alignments-load-precomputed-model-py" title="Permalink to this headline">¶</a></h3>
<p>The script takes one fasta alignment and its associated tree and runs all the models specified by the user</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>   # command line :
   ./compCodeML_ete_oneFile.py /path/to/phylogenetic/tree/used/in/codeML /path/to/super/alignment/used/in/codeML list,of,models,to,run,comma,separated

- The tree computed on the super-alignment can be used, but you can also run RAxML on each alignement to have one tree per alignement
- It is planned to run this script in parallel on many input files.
- The models names must be as written in ete3 (see table below).
</pre></div>
</div>
</div>
</div>
<div class="section" id="ete-available-models-and-comparisons">
<h2>ETE available models and comparisons:<a class="headerlink" href="#ete-available-models-and-comparisons" title="Permalink to this headline">¶</a></h2>
<div class="section" id="models-implemented-in-ete3-module">
<h3>Models implemented in ete3 module<a class="headerlink" href="#models-implemented-in-ete3-module" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Available models</span>

<span class="c1"># Name   | Description                 | Model kind</span>
<span class="c1"># ------------------------------------------------</span>
<span class="c1"># M1     | relaxation                  | site</span>
<span class="c1"># M10    | beta and gamma + 1          | site</span>
<span class="c1"># M11    | beta and normal &gt; 1         | site</span>
<span class="c1"># M12    | 0 and 2 normal &gt; 2          | site</span>
<span class="c1"># M13    | 3 normal &gt; 0                | site</span>
<span class="c1"># M2     | positive-selection          | site</span>
<span class="c1"># M3     | discrete                    | site</span>
<span class="c1"># M4     | frequencies                 | site</span>
<span class="c1"># M5     | gamma                       | site</span>
<span class="c1"># M6     | 2 gamma                     | site</span>
<span class="c1"># M7     | relaxation                  | site</span>
<span class="c1"># M8     | positive-selection          | site</span>
<span class="c1"># M8a    | relaxation                  | site</span>
<span class="c1"># M9     | beta and gamma              | site</span>
<span class="c1"># SLR    | positive/negative selection | site</span>
<span class="c1"># M0     | negative-selection          | null</span>
<span class="c1"># fb_anc | free-ratios                 | branch_ancestor</span>
<span class="c1"># bsA    | positive-selection          | branch-site</span>
<span class="c1"># bsA1   | relaxation                  | branch-site</span>
<span class="c1"># bsB    | positive-selection          | branch-site</span>
<span class="c1"># bsC    | different-ratios            | branch-site</span>
<span class="c1"># bsD    | different-ratios            | branch-site</span>
<span class="c1"># b_free | positive-selection          | branch</span>
<span class="c1"># b_neut | relaxation                  | branch</span>
<span class="c1"># fb     | free-ratios                 | branch</span>
<span class="c1"># XX     | User defined                | Unknown</span>
</pre></div>
</div>
</div>
<div class="section" id="usual-comparisons">
<h3>Usual comparisons<a class="headerlink" href="#usual-comparisons" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Usual comparisons are :</span>

<span class="c1"># ============ ======= ===========================================</span>
<span class="c1">#  Alternative  Null    Test</span>
<span class="c1"># ============ ======= ===========================================</span>
<span class="c1">#   M2          M1      PS on sites (M2 prone to miss some sites)</span>
<span class="c1">#                       (Yang 2000).</span>
<span class="c1">#   M3          M0      test of variability among sites</span>
<span class="c1">#   M8          M7      PS on sites</span>
<span class="c1">#                       (Yang 2000)</span>
<span class="c1">#   M8          M8a     RX on sites?? think so....</span>
<span class="c1">#   bsA         bsA1    PS on sites on specific branch</span>
<span class="c1">#                       (Zhang 2005)</span>
<span class="c1">#   bsA         M1      RX on sites on specific branch</span>
<span class="c1">#                       (Zhang 2005)</span>
<span class="c1">#   bsC         M1      different omegas on clades branches sites</span>
<span class="c1">#                       ref: Yang Nielsen 2002</span>
<span class="c1">#   bsD         M3      different omegas on clades branches sites</span>
<span class="c1">#                       (Yang Nielsen 2002, Bielawski 2004)</span>
<span class="c1">#   b_free      b_neut  foreground branch not neutral (w != 1)</span>
<span class="c1">#                        - RX if P&lt;0.05 (means that w on frg=1)</span>
<span class="c1">#                        - PS if P&gt;0.05 and wfrg&gt;1</span>
<span class="c1">#                        - CN if P&gt;0.05 and wfrg&gt;1</span>
<span class="c1">#                        (Yang Nielsen 2002)</span>
<span class="c1">#   b_free      M0      different ratio on branches</span>
<span class="c1">#                       (Yang Nielsen 2002)</span>
<span class="c1"># ============ ======= ===========================================</span>
<span class="c1"># **Note that M1 and M2 models are making reference to the new versions</span>
<span class="c1"># of these models, with continuous omega rates (namely M1a and M2a in the</span>
<span class="c1"># PAML user guide).**</span>

<span class="c1"># **Alternative must have a greater number of parameters than Null**</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The tables above are also written in the Python scripts.</p>
</div>
</div>
</div>
<div class="section" id="code-documentation">
<h2>Code documentation<a class="headerlink" href="#code-documentation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="functions-for-the-detection-of-positive-selection-functions-positive-selection-py">
<h3>Functions for the detection of positive selection : functions_positive_selection.py<a class="headerlink" href="#functions-for-the-detection-of-positive-selection-functions-positive-selection-py" title="Permalink to this headline">¶</a></h3>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">These functions are not finished ! (implementation have been stopped for a while due to versions and dependancies conflicts).</p>
</div>
<dl class="function">
<dt id="details_on_sites">
<code class="descname">details_on_sites</code><span class="sig-paren">(</span><em>model</em>, <em>tree</em>, <em>models_types</em>, <em>verbose</em><span class="sig-paren">)</span><a class="headerlink" href="#details_on_sites" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>model</strong> (<em>ete3 model object</em>) – must be a site-model</li>
<li><strong>tree</strong> (<em>ete3 tree object</em>) – the tree used for running models</li>
<li><strong>models_types</strong> – the list of evolutionary models, sorted by type</li>
<li><strong>verbose</strong> (<em>bool</em>) – activate the display of conserved and neutral sites in the result</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>This function is for printing the location of the positively selected sites. Under model M1, there are no such sites (only w &lt; 1 and w = 1)</p>
<dl class="function">
<dt id="frame_site">
<code class="descname">frame_site</code><span class="sig-paren">(</span><em>model</em>, <em>tree</em>, <em>models_types</em>, <em>verbose</em><span class="sig-paren">)</span><a class="headerlink" href="#frame_site" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>model</strong> (<em>ete3 model object</em>) – must be a site-model</li>
<li><strong>tree</strong> (<em>ete3 tree object</em>) – the tree used for running models</li>
<li><strong>models_types</strong> – the list of evolutionary models, sorted by type</li>
<li><strong>verbose</strong> (<em>bool</em>) – activate the display of conserved and neutral sites in the result</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<div class="admonition-todo admonition" id="index-3">
<p class="first admonition-title">Todo</p>
<p class="last">This function works in the same way than details_on_sites() but writes the results in a csv table.</p>
</div>
<dl class="function">
<dt id="details_on_branches">
<code class="descname">details_on_branches</code><span class="sig-paren">(</span><em>pval</em>, <em>alt_model</em>, <em>null_model</em>, <em>tree</em><span class="sig-paren">)</span><a class="headerlink" href="#details_on_branches" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>model</strong> (<em>ete3 model object</em>) – must be a site-model</li>
<li><strong>tree</strong> (<em>ete3 tree object</em>) – the tree used for running models</li>
<li><strong>models_types</strong> – the list of evolutionary models, sorted by type</li>
<li><strong>verbose</strong> (<em>bool</em>) – activate the display of conserved and neutral sites in the result</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<div class="admonition-todo admonition" id="index-4">
<p class="first admonition-title">Todo</p>
<p class="last">This function is for printing positively selected branches on screen</p>
</div>
<dl class="function">
<dt id="models_types">
<code class="descname">models_types</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#models_types" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Return m_types:</th><td class="field-body">the list of models classified by their type (branch/site/branch-site/branch-ancestor/null)</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>
<div class="section" id="conda-environment">
<h2>Conda environment<a class="headerlink" href="#conda-environment" title="Permalink to this headline">¶</a></h2>
<p>I used a conda environment (with Python 2.7) :</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span># Create environment with Python 2
conda create -n env_ete3 python=2.7
source activate env_ete3

# Packages which a specific version
conda install -c anaconda qt=4.8.7
conda install -c anaconda sip=4.18
conda install -c anaconda pyqt=4.11.4

# Pandas and numpy
conda install -c anaconda numpy=1.15
conda install -c anaconda pandas=0.23.3

# ete3 and paml
conda install -c etetoolkit ete3
conda install -c bioconda paml

# Other dependancies will be automatically installed
</pre></div>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">This has been already mentionned, but we had many versions and dependancies conflicts and crashes ! I suspect a conflict between Python versions (2.7/3.4) / PyQt versions (4/5) and ete3 versions. This needs to be investigated. The specified versions of the packages cited above will, <em>apriori</em>, avoid crashes.</p>
</div>
</div>
<div class="section" id="reference">
<h2>Reference<a class="headerlink" href="#reference" title="Permalink to this headline">¶</a></h2>
<p>ETE 3: Reconstruction, analysis and visualization of phylogenomic data. Jaime Huerta-Cepas, Francois Serra and Peer Bork. Mol Biol Evol 2016; doi: 10.1093/molbev/msw046</p>
<p>Website : <a class="reference external" href="http://etetoolkit.org">http://etetoolkit.org</a></p>
<p>Back to <a class="reference external" href="index.html">main page</a>.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Filter_Assemblies.html">Filter_Assemblies</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pairwise.html">Pairwise</a></li>
<li class="toctree-l1"><a class="reference internal" href="POGs.html">POGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BlastAlign.html">BlastAlign</a></li>
<li class="toctree-l1"><a class="reference internal" href="TransDecoder.html">TransDecoder</a></li>
<li class="toctree-l1"><a class="reference internal" href="OrthoFinder_OnlyGroups.html">OrthoFinder_OnlyGroups</a></li>
<li class="toctree-l1"><a class="reference internal" href="Orthogroups_Tool.html">Orthogroups_Tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="CDS_Search.html">CDS_Search</a></li>
<li class="toctree-l1"><a class="reference internal" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li class="toctree-l1"><a class="reference internal" href="MutCount.html">MutCount</a></li>
<li class="toctree-l1"><a class="reference internal" href="CodeML.html">codeML</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">GeneSearch</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#handling-of-codeml-runs-on-a-super-alignment-from-adaptsearch-s-galaxy-instance">Handling of codeML runs on a super-alignment from AdaptSearch’s Galaxy instance</a></li>
<li class="toctree-l3"><a class="reference internal" href="#running-an-evolutionary-model-on-many-differents-alignments-at-once">Running an evolutionary model on many differents alignments at once :</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#usage">Usage</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#images-outputs-or-gui-s-visualisation">Images outputs or GUI’s visualisation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#script-for-exploitation-of-codeml-results-load-precomputed-model-py">Script for exploitation of codeML results : load_precomputed_model.py</a></li>
<li class="toctree-l3"><a class="reference internal" href="#script-for-running-codeml-models-on-one-or-many-alignments-load-precomputed-model-py">Script for running codeML models on one or many alignments : load_precomputed_model.py</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ete-available-models-and-comparisons">ETE available models and comparisons:</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#models-implemented-in-ete3-module">Models implemented in ete3 module</a></li>
<li class="toctree-l3"><a class="reference internal" href="#usual-comparisons">Usual comparisons</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#code-documentation">Code documentation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#functions-for-the-detection-of-positive-selection-functions-positive-selection-py">Functions for the detection of positive selection : functions_positive_selection.py</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#conda-environment">Conda environment</a></li>
<li class="toctree-l2"><a class="reference internal" href="#reference">Reference</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Visualization_With_Python.html">Visualization with Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Shiny_Apps.html">Visualization with RShiny apps</a></li>
<li class="toctree-l1"><a class="reference internal" href="CommandLine_User_Guide.html">User Guide for the use of AdaptSearch in command line</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="CodeML.html"
                        title="previous chapter">codeML</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="Visualization_With_Python.html"
                        title="next chapter">Visualization with Python</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/GeneSearch.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="Visualization_With_Python.html" title="Visualization with Python"
             >next</a> |</li>
        <li class="right" >
          <a href="CodeML.html" title="codeML"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Alice Mataigne.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.1.
    </div>
  </body>
</html>
