
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>MutCount &#8212; AdaptSearch 2 documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="codeML" href="CodeML.html" />
    <link rel="prev" title="ConcatPhyl" href="ConcatPhyl.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="CodeML.html" title="codeML"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="ConcatPhyl.html" title="ConcatPhyl"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="mutcount">
<h1>MutCount<a class="headerlink" href="#mutcount" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<div class="section" id="separated-mode">
<h3>Separated mode<a class="headerlink" href="#separated-mode" title="Permalink to this headline">¶</a></h3>
<dl class="docutils">
<dt>This mode counts occurrences of amino-acids or nucleic acids, according to the sequences type, in each distinct orthogroup. Then, two subgroups of species are set by the user :</dt>
<dd><ol class="first last arabic simple">
<li>A first group, constitued by species having something in common (an ecological trait, an ecological niche, a particular environmental adaptation)</li>
<li>A second group, constitued by species sharing the opposite trait (for example, the user can have a first subgroup made with species adapted to high temperatures and a second group made with species adapted to cold temperatures)</li>
</ol>
</dd>
</dl>
<p>Within the groups, the program checks wether the occurrences of each element (amino-acid, nucleic acid, thermostability indice, GC content …) is higher of lower between one species and all the species of the opposite group. Binomial tests are then performed of these counts.</p>
</div>
<div class="section" id="concatenated-mode">
<h3>Concatenated mode<a class="headerlink" href="#concatenated-mode" title="Permalink to this headline">¶</a></h3>
<p>This script counts the number of codons, amino acids, and types of amino acids in sequences, as well as the mutation bias from one item to another between 2 sequences. Counting is then compared to empirical p-values, obtained from bootstrapped sequences obtained from a subset of sequences.</p>
<p>In the output files, the pvalues indicate the position of the observed data in a distribution of empirical countings obtained from a resample of the data. Values above 0.95 indicate a significantly higher counting, values under 0.05 a significantly lower counting.</p>
<p>The script resamples random pairs of aligned codon to determine what countings can be expected under the hypothesis of an homogenous dataset. Countings are performed on each generated random alignement, thousands of alignments allow to draw a gaussian distribution of the countings. Then the script simply checks whether the observed data are within the 5% lowest or 5% highest values of the distribution.</p>
</div>
</div>
<div class="section" id="user-parameters">
<h2>User parameters<a class="headerlink" href="#user-parameters" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id1">
<h3>Separated mode<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><ol class="arabic simple">
<li>Sequences type has to be specified : ‘nucleic’ or ‘proteic’</li>
<li>Inputs are all fasta files from <a class="reference external" href="CDS_Search.html">CDS_Search</a> read from a fixed named directory of inputs</li>
<li>List of all species : the user has to specifiy all the species which can be found in the set of orthogroups (comma-separated abbreviated names, ex Ac,Ap,Ps,Pp,Pf,Am,Pg)</li>
<li>Species for group one : set the list of species of group one (comma-separated abbreviated names, ex Ap,Ps)</li>
<li>Species for group two : set the list of species of group two (comma-separated abbreviated names, ex Pg,Pp)</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="id2">
<h3>Concatenated mode<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><ol class="arabic simple">
<li>Input fasta file is the ‘Phylogeny_concatenation_fasta_nuc’ from <a class="reference external" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li>List of species for countings : the user can focus countings on these species only</li>
<li>List of species used for resampling : the user can focus the resampling on these species only</li>
<li>Number of sampled codons : sets the length (in codons) of the resampled sequences</li>
<li>Number of iterations : sets the number of resampled sequences</li>
</ol>
</div></blockquote>
</div>
</div>
<div class="section" id="code-documentation">
<h2>Code documentation<a class="headerlink" href="#code-documentation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="algorithm">
<h3>Algorithm<a class="headerlink" href="#algorithm" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id3">
<h4>Separated mode<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><ol class="arabic simple">
<li>A first script proceed to count the occurrence of each element (nucleotide or amino-acid) in each orthogroup, then of each indice (GC and purine content, IVYWREL, VLIM, hydratation …). Results are stored in nested dictionaries (keys : files_names, sub-keys : species names, sub-sub-keys : elements and indices names). Missing species in a group are set with dictionaries full of ‘NaN’ values. One cs file is written per element/indice.</li>
<li>A second script prepare tables for binomial test by counting the number of successes and failures between the two subgroups of species (e.g the number of times a value of A is higher (and lower) than a value of B).</li>
<li>A species is compared to the whole group of opposite species -&gt; the maximum and minimum values of ‘A’ observed into the opposite group are taken into account for the comparisons</li>
<li>A third R script loops on these tables to do binomial tests. The number of successes is, for instance, the number of times the value of ‘A’ has been higher than the value of ‘B’, and the number of trials is the number of times ‘A’ &gt; ‘B’ + the number of times ‘A’ &lt; ‘B’</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This is not how a binomial test is supposed to work. The number of trials should be the total number of comparisons (‘A’ &gt; ‘B’ + ‘A’&lt;’B’ + ‘A’=’B’). Is it done on purpose ?</p>
</div>
</div></blockquote>
</div>
<div class="section" id="id4">
<h4>Concatenated mode<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h4>
<dl class="docutils">
<dt>A whole bunch of dictionaries are used to store, for each sequences/pairs of sequences :</dt>
<dd><ol class="first last arabic simple">
<li>Each codons/amino-acids/amino-acids types occurrences countings/frequences (3*2=6 dictionaries)</li>
<li>Each codons/amino-acids/amino-acids types transitions countings/frequences (3*2=6 dictionaries)</li>
</ol>
</dd>
</dl>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Example</span>
<span class="n">codons_counts_species_X</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="o">...</span><span class="p">}</span>
<span class="n">codons_transitions_counts_species_X_to_species_Y</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                    <span class="s1">&#39;aac&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                    <span class="o">...</span><span class="p">}</span>
</pre></div>
</div>
<p>Resampling is done on all or on a subset of sequences by picking codons at random positions (which are positions where all the resampled sequences have a full codon).</p>
<p>Then, Final dictionaries gathers all occurrences and transitions of all sequences, which are written to csv files using pandas.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Example</span>
<span class="n">codons_counts_final</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Species1&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">20</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                       <span class="s1">&#39;Species2&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">14</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">17</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                       <span class="o">...</span><span class="p">}</span>
<span class="n">codons__transitions_counts_final</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;species_X_to_species_Y&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="s1">&#39;aac&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">15</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="o">...</span><span class="p">}</span>
                                    <span class="s1">&#39;species_X_to_species_Z&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="s1">&#39;aac&#39;</span><span class="p">:{</span><span class="s1">&#39;aaa&#39;</span><span class="p">:</span><span class="mi">3</span><span class="p">,</span> <span class="s1">&#39;aac&#39;</span><span class="p">:</span><span class="mi">10</span><span class="p">,</span> <span class="o">...</span><span class="p">},</span>
                                                               <span class="o">...</span><span class="p">}</span>
                                    <span class="o">...</span><span class="p">}</span>
</pre></div>
</div>
<p>Various values (IVRYWEL, EKQH, PAYRESDGM, GC12 …) are stored in int/float variables, put in a list (one per species) and then also written in csv file.</p>
</div>
</div>
<div class="section" id="source-code">
<h3>Source code<a class="headerlink" href="#source-code" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id5">
<h4>Separated mode<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h4>
<div class="section" id="file-s01b-extract-variable-nuc-py">
<h5>File: S01b_extract_variable_nuc.py<a class="headerlink" href="#file-s01b-extract-variable-nuc-py" title="Permalink to this headline">¶</a></h5>
<p>Counts occurrences of nucleic acids and computes corresponding indices (GC content, purine content …). Called when the sequences are in nucleic format.</p>
<dl class="function">
<dt id="all_nuc_counts">
<code class="descname">all_nuc_counts</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="headerlink" href="#all_nuc_counts" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> (<em>String</em>) – a nucleic sequence</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">counts of each nucleotide</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ratios">
<code class="descname">ratios</code><span class="sig-paren">(</span><em>nuc_counts</em><span class="sig-paren">)</span><a class="headerlink" href="#ratios" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>nuc_counts</strong> (<em>dict</em>) – counts of nucleotides in a sequence (computed by all_nuc_counts())</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">values of various nucleic indices and ratios (GC%, purine% …)</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>For each element (A,C,G,T in a first directory, and each indice in a second directory), there is an output csv file. Rows are fasta files and columns are species. Missing species in a group have NaN values.</p>
</div>
<div class="section" id="file-s01b-extract-variable-prot-py">
<h5>File: S01b_extract_variable_prot.py<a class="headerlink" href="#file-s01b-extract-variable-prot-py" title="Permalink to this headline">¶</a></h5>
<p>Counts occurrences of amino acids and computes corresponding indices (IVYRWEL, ratio EQ_KQ …). Called when the sequences are in proteic format.</p>
<dl class="function">
<dt id="all_aa_counts">
<code class="descname">all_aa_counts</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="headerlink" href="#all_aa_counts" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> (<em>String</em>) – a proteic sequence</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">counts of each amino-acid</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="all_aa_props">
<code class="descname">all_aa_props</code><span class="sig-paren">(</span><em>seq_counts</em><span class="sig-paren">)</span><a class="headerlink" href="#all_aa_props" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq_counts</strong> (<em>dict</em>) – counts of amino-acids in a sequence (computed by all_aa_counts())</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">proportion of each amino-acid in the sequence</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>This function is called but is curently excluded from the final results.</p>
<dl class="function">
<dt id="aa_variables_counts_and_props">
<code class="descname">aa_variables_counts_and_props</code><span class="sig-paren">(</span><em>aa_counts</em><span class="sig-paren">)</span><a class="headerlink" href="#aa_variables_counts_and_props" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aa_counts</strong> (<em>dict</em>) – counts of amino-acids in a sequence (computed by all_aa_counts())</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">2 variables : counts and proportions of various indices and ratios</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="aa_properties">
<code class="descname">aa_properties</code><span class="sig-paren">(</span><em>amino_acids_properties_file</em><span class="sig-paren">)</span><a class="headerlink" href="#aa_properties" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>amino_acids_properties_file</strong> (<em>String</em>) – a file which comes along with the scripts, storing various amino-acids properties</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">the content of the file, each amino-acid properties being a list (frequencies, weight, volume, partial_specific_volume, hydratation)</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="sequence_properties_from_aa_properties">
<code class="descname">sequence_properties_from_aa_properties</code><span class="sig-paren">(</span><em>aa_counts</em>, <em>aa_properties</em><span class="sig-paren">)</span><a class="headerlink" href="#sequence_properties_from_aa_properties" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>aa_counts</strong> (<em>dict</em>) – counts of amino-acids in a sequence (computed by all_aa_counts())</li>
<li><strong>aa_properties</strong> (<em>dict</em>) – amino-acids properties returned by the call of aa_properties()</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>For each element (each amino-acid in a first directory, and each indice in a second directory), there is an output csv file. Rows are fasta files and columns are species. Missing species in a group have NaN values.</p>
</div>
<div class="section" id="file-s02b-extreme-2states-py">
<h5>File: S02b_extreme_2states.py<a class="headerlink" href="#file-s02b-extreme-2states-py" title="Permalink to this headline">¶</a></h5>
<p>Proceeds to count, for each element/indice and each comparison <em>species A from group 1</em> VS. <em>all species from group 2</em>, the number of times A &gt; B and A &lt; B. The script handles both sequences format (nucleic and proteic).</p>
<dl class="function">
<dt id="loop_on_elems">
<code class="descname">loop_on_elems</code><span class="sig-paren">(</span><em>list_of_elems</em>, <em>path_in</em>, <em>path_out</em>, <em>sps_group_1</em>, <em>sps_group_2</em>, <em>colnames</em><span class="sig-paren">)</span><a class="headerlink" href="#loop_on_elems" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>list_of_elems</strong> (<em>list</em>) – the lsit of counted elements names, which corresponds to dictionaries keys</li>
<li><strong>path_in</strong> (<em>String</em>) – The name of the input directory</li>
<li><strong>path_out</strong> (<em>String</em>) – The name of the output directory</li>
<li><strong>sps_group_1</strong> (<em>list</em>) – Species names (abbreviated) belonging to group 1</li>
<li><strong>sps_group_2</strong> (<em>list</em>) – Species names (abbreviated) belonging to group 2</li>
<li><strong>colnames</strong> (<em>dict</em>) – contains automaticcaly formatted names for the table columns (one per comparison)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>This is the main function, which loops on every counts file, make pandas dataframe which are then written in csv files. It calls the sub-routine tableu().</p>
<dl class="function">
<dt id="tableu">
<code class="descname">tableu</code><span class="sig-paren">(</span><em>fileu</em>, <em>sps_group_1</em>, <em>sps_group_2</em><span class="sig-paren">)</span><a class="headerlink" href="#tableu" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fileu</strong> (<em>pandas DataFrame</em>) – contains the counts of an element computed by the previous script</li>
<li><strong>sps_group_1</strong> (<em>list</em>) – Same parameter than into the function loop_on_elems()</li>
<li><strong>sps_group_2</strong> (<em>list</em>) – Same parameter than into the function loop_on_elems()</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>This function parse the pandas DataFrame obtained after reading the input counts file and proceeds to count the number of successes for A&lt;B and A&gt;B</p>
<p>At the end of this second script, a R script (S03b_sign_test_binomial.R) is called twice (on elements counts and on indices counts) and run binomial tests.</p>
</div>
</div>
<div class="section" id="id6">
<h4>Concatenated mode<a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h4>
<dl class="function">
<dt id="buildDicts">
<code class="descname">buildDicts</code><span class="sig-paren">(</span><em>list_codons</em>, <em>content</em>, <em>dict_genetic_code</em>, <em>dict_aa_classif</em><span class="sig-paren">)</span><a class="headerlink" href="#buildDicts" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>list_codons</strong> (<em>list</em>) – all codons except stop-codons</li>
<li><strong>content</strong> (<em>int</em><em> (</em><em>for countings</em><em>) or </em><em>list</em><em> (</em><em>for resampling</em><em>)</em>) – specifies the keys type</li>
<li><strong>dict_genetic_code</strong> (<em>dict</em>) – Describe the genetic code (key : amino-acid, values : list of codons)</li>
<li><strong>dict_aa_classif</strong> (<em>dict</em>) – Describe the amino-acids classification (key : amino-acid type, values : list of amino-acids)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 dictionaries used for countings and frequencies storage on a species or resampling</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>At each species or pair of species countings, a copy of the input dictionaries is done, and dictionaries for transitions are also built.</p>
<dl class="function">
<dt id="viable">
<code class="descname">viable</code><span class="sig-paren">(</span><em>seqs</em>, <em>pos</em>, <em>method</em><span class="sig-paren">)</span><a class="headerlink" href="#viable" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seqs</strong> (<em>list</em>) – sequences, which must all have the same size</li>
<li><strong>pos</strong> (<em>int</em>) – positions, &lt;= len(seqs) -3</li>
<li><strong>method</strong> (<em>string</em>) – “all” or “any”</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">if the given position is viable in all the sequence</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">bool</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Compute if, among a set of sequences, either at least one of the codons at the specified position has not a “-“, or not any codon has a “-“</p>
<dl class="function">
<dt id="computeAllCountingsAndFreqs">
<code class="descname">computeAllCountingsAndFreqs</code><span class="sig-paren">(</span><em>seq</em>, <em>list_codons</em>, <em>init_dict_codons</em>, <em>init_dict_aa</em>, <em>init_dict_classif</em>, <em>dict_genetic_code</em>, <em>dict_aa_classif</em><span class="sig-paren">)</span><a class="headerlink" href="#computeAllCountingsAndFreqs" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>list_codons</strong> (<em>list</em>) – all codons except stop-codons</li>
<li><strong>init_dict_codons</strong> (<em>dict</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>init_dict_aa</strong> (<em>dict</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>init_dict_classif</strong> (<em>dict</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>dict_genetic_code</strong> (<em>dict</em>) – Describe the genetic code (key : amino-acid, values : list of codons)</li>
<li><strong>dict_aa_classif</strong> (<em>dict</em>) – Describe the amino-acids classification (key : amino-acid type, values : list of amino-acids)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 variables : countings and frequencies of codons, amino-acid and amino-acid types</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Call all functions dedicated to the computation of occurences and frequencies of codons, amino-acids, amino-acids types</p>
<dl class="function">
<dt id="computeVarious">
<code class="descname">computeVarious</code><span class="sig-paren">(</span><em>seq</em>, <em>dict_aa_counts</em>, <em>dict_aa_types</em><span class="sig-paren">)</span><a class="headerlink" href="#computeVarious" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>dict_aa_counts</strong> (<em>dict</em>) – the countings of amino-acids on the param <cite>seq</cite></li>
<li><strong>dict_aa_types</strong> (<em>dict</em>) – the countings of amino-acids types on the param <cite>seq</cite></li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">GC3, GC12, IVYWREL, EKQH, PAYRESDGM, purineload, CvP values for <cite>seq</cite></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int and floats</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Call al the functions for nucleic and amino-acids sequences description : <em>GC3, GC12, IVYWREL, EKQH, PAYRESDGM, purineload, CvP</em></p>
<dl class="function">
<dt id="computeAllBiases">
<code class="descname">computeAllBiases</code><span class="sig-paren">(</span><em>seq1</em>, <em>seq2</em>, <em>dico_codons_transi</em>, <em>dico_aa_transi</em>, <em>dico_aatypes_transi</em>, <em>reversecode</em>, <em>reverseclassif</em><span class="sig-paren">)</span><a class="headerlink" href="#computeAllBiases" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seq1</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>seq2</strong> (<em>string</em>) – a dna/rna sequence (a,t,c,g)</li>
<li><strong>dico_codons_transi</strong> (<em>dict of dicts</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>dico_aa_transi</strong> (<em>dict of dicts</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>dico_aatypes_transi</strong> (<em>dict of dicts</em>) – a copied output of <cite>buildDicts()</cite></li>
<li><strong>reversecode</strong> (<em>dict</em>) – the genetic_code, but reversed (key : codon, values : amino-acid)</li>
<li><strong>reverseclassif</strong> (<em>dict</em>) – the amino-acids classification, but reversed (key : amino-acid, values : amino-acid type)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">countings and frequencies of codons, amino-acid and amino-acid types transitions from <cite>seq1</cite> to <cite>seq2</cite></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict of dicts</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Compute all biases (transisitions codon-&gt;codon, aa-&gt;-aa, aa_type-&gt;aa_type) between two sequences</p>
<dl class="function">
<dt id="sampling">
<code class="descname">sampling</code><span class="sig-paren">(</span><em>dict_seq</em>, <em>nb_iter</em>, <em>len_sample</em>, <em>list_codons</em>, <em>genetic_code</em>, <em>aa_classif</em>, <em>reversecode</em>, <em>reverseclassif</em><span class="sig-paren">)</span><a class="headerlink" href="#sampling" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dict_seq</strong> (<em>dict</em>) – contains the species name (keys) and sequences (values) used</li>
<li><strong>nb_iter</strong> (<em>int</em>) – the number of iterations</li>
<li><strong>len_sample</strong> (<em>int</em>) – the lenght (in codons) of each resampled sequence</li>
<li><strong>list_codons</strong> (<em>list</em>) – all codons except stop-codons</li>
<li><strong>genetic_code</strong> (<em>dict</em>) – Describe the genetic code (key : amino-acid, values : list of codons)</li>
<li><strong>aa_classif</strong> (<em>dict</em>) – Describe the amino-acids classification (key : amino-acid type, values : list of amino-acids)</li>
<li><strong>reversecode</strong> (<em>dict</em>) – the genetic_code, but reversed (key : codon, values : amino-acid)</li>
<li><strong>reverseclassif</strong> (<em>dict</em>) – the amino-acids classification, but reversed (key : amino-acid, values : amino-acid type)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">6 variables</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict and dict of dicts</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Resample randomly codons from sequences (sort of bootsrap)</p>
<dl class="function">
<dt id="testPvalues">
<code class="descname">testPvalues</code><span class="sig-paren">(</span><em>dict_counts</em>, <em>dict_resampling</em>, <em>nb_iter</em><span class="sig-paren">)</span><a class="headerlink" href="#testPvalues" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dict_counts</strong> (<em>dict</em>) – contains codons frequencies of a sequence, an output of <cite>computeAllCountingsAndFreqs()</cite></li>
<li><strong>dict_resampling</strong> (<em>dict</em>) – </li>
<li><strong>nb_iter</strong> (<em>int</em>) – the number of iterrations used for resampling</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">pvalues for each item (codon, amino-acid, amino-acid type countings or transitions)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">dict</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Computes where the observed value is located in the expected counting distribution</p>
<p>Back to <a class="reference external" href="index.html">main page</a>.</p>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Filter_Assemblies.html">Filter_Assemblies</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pairwise.html">Pairwise</a></li>
<li class="toctree-l1"><a class="reference internal" href="POGs.html">POGs</a></li>
<li class="toctree-l1"><a class="reference internal" href="BlastAlign.html">BlastAlign</a></li>
<li class="toctree-l1"><a class="reference internal" href="TransDecoder.html">TransDecoder</a></li>
<li class="toctree-l1"><a class="reference internal" href="OrthoFinder_OnlyGroups.html">OrthoFinder_OnlyGroups</a></li>
<li class="toctree-l1"><a class="reference internal" href="Orthogroups_Tool.html">Orthogroups_Tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="CDS_Search.html">CDS_Search</a></li>
<li class="toctree-l1"><a class="reference internal" href="ConcatPhyl.html">ConcatPhyl</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">MutCount</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#separated-mode">Separated mode</a></li>
<li class="toctree-l3"><a class="reference internal" href="#concatenated-mode">Concatenated mode</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#user-parameters">User parameters</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id1">Separated mode</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id2">Concatenated mode</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#code-documentation">Code documentation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#algorithm">Algorithm</a></li>
<li class="toctree-l3"><a class="reference internal" href="#source-code">Source code</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="CodeML.html">codeML</a></li>
<li class="toctree-l1"><a class="reference internal" href="GeneSearch.html">GeneSearch</a></li>
<li class="toctree-l1"><a class="reference internal" href="Visualization_With_Python.html">Visualization with Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Shiny_Apps.html">Visualization with RShiny apps</a></li>
<li class="toctree-l1"><a class="reference internal" href="CommandLine_User_Guide.html">User Guide for the use of AdaptSearch in command line</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="ConcatPhyl.html"
                        title="previous chapter">ConcatPhyl</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="CodeML.html"
                        title="next chapter">codeML</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/MutCount.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="CodeML.html" title="codeML"
             >next</a> |</li>
        <li class="right" >
          <a href="ConcatPhyl.html" title="ConcatPhyl"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">AdaptSearch 2 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Alice Mataigne.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.1.
    </div>
  </body>
</html>
